File: src/main/java/aurocosh/divinefavor/common/item/talismans/spell/SpellTalismanWarpMarker.java
package aurocosh.divinefavor.common.item.talismans.spell;

import aurocosh.divinefavor.common.item.ItemWarpMarker;
import aurocosh.divinefavor.common.item.base.ModItem;
import aurocosh.divinefavor.common.item.talismans.spell.base.ItemSpellTalisman;
import aurocosh.divinefavor.common.item.talismans.spell.base.SpellOptions;
import aurocosh.divinefavor.common.item.talismans.spell.base.TalismanContext;
import aurocosh.divinefavor.common.spirit.base.ModSpirit;
import aurocosh.divinefavor.common.util.UtilNbt;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NBTTagCompound;

import java.util.EnumSet;

public class SpellTalismanWarpMarker extends ItemSpellTalisman {
    private final ModItem item;

    public SpellTalismanWarpMarker(String name, ModSpirit spirit, int favorCost, EnumSet<SpellOptions> options, ModItem item) {
        super(name, spirit, favorCost, options);
        this.item = item;
    }

    @Override
    protected void performActionServer(TalismanContext context) {
        EntityPlayer player = context.player;
        ItemStack stack = new ItemStack(item);
        NBTTagCompound nbt = UtilNbt.getNbt(stack);
        UtilNbt.setBlockPos(nbt, ItemWarpMarker.TAG_POSITION, player.getPosition());
        nbt.setInteger(ItemWarpMarker.TAG_DIMENSION, player.dimension);
        player.inventory.addItemStackToInventory(stack);
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/item/talismans/spell/SpellTalismanWarpMarker.kt
package aurocosh.divinefavor.common.item.talismans.spell

import aurocosh.divinefavor.common.item.base.ModItem
import aurocosh.divinefavor.common.item.gems.ItemWarpMarker
import aurocosh.divinefavor.common.item.talismans.spell.base.ItemSpellTalisman
import aurocosh.divinefavor.common.item.talismans.spell.base.SpellOptions
import aurocosh.divinefavor.common.item.talismans.spell.base.TalismanContext
import aurocosh.divinefavor.common.lib.extensions.compound
import aurocosh.divinefavor.common.lib.extensions.setBlockPos
import aurocosh.divinefavor.common.spirit.base.ModSpirit
import net.minecraft.item.ItemStack
import java.util.*

class SpellTalismanWarpMarker(name: String, spirit: ModSpirit, favorCost: Int, options: EnumSet<SpellOptions>, private val item: ModItem) : ItemSpellTalisman(name, spirit, favorCost, options) {

    override fun performActionServer(context: TalismanContext) {
        val player = context.player
        val stack = ItemStack(item)
        val compound = stack.compound
        compound.setBlockPos(ItemWarpMarker.TAG_POSITION, player.position)
        compound.setInteger(ItemWarpMarker.TAG_DIMENSION, player.dimension)
        player.inventory.addItemStackToInventory(stack)
    }
}
